nav2_smac_planner/include/nav2_smac_planner/goal_manager.hpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2020, Samsung Research America
```

Comment:
Make this a hearer only utility (i.e. move the CPP implementations here). That, I think, architecturally makes it clear that this is a utility and not a main part of the project's files that someone just getting started with the project will look at.

Reply:
There's also alot of functions here that are being marked as lacking test coverage. Please write some unit tests for this that covers all the APIs -- in order of how they are used in the A* to mimic real-world use

Reply:
Done